java/org/contikios/cooja/Simulation.java

```diff
@@ -222,24 +291,8 @@ public void run() {
    * @param r Simulation thread action
    */
   public void invokeSimulationThread(Runnable r) {
-    synchronized (pollRequests) {
-      pollRequests.addLast(r);
-      hasPollRequests = true;
-    }
-  }
-
-  private void runSimulationInvokes() {
-    boolean more;
-    synchronized (pollRequests) {
-      more = hasPollRequests;
-    }
-    while (more) {
-      Runnable r;
-      synchronized (pollRequests) {
-        r = pollRequests.pop();
-        more = hasPollRequests = !pollRequests.isEmpty();
-      }
-      r.run();
+    if (!isShutdown) {
+      this.commandQueue.add(r);
```

Comment:
Don't think `this.` adds anything in this context.

Reply:
No, it does not. Old habit from another code style. I will remove it.